spec/decors_spec.rb

```diff
@@ -12,10 +12,37 @@ def self.arguments(*); end
         }
     }
 
+    context 'Check decorated_method & undecorated_method' do
+        let(:instance) { TestClass.new }
+
+        before {
+            stub_class(:SimpleDecorator, inherits: [::Decors::DecoratorBase])
+            call_method = call_proc
+            SimpleDecorator.send(:define_method, :call){ |*| instance_eval(&call_method) }
+            TestClass.class_eval {
+                define_decorator :SimpleDecorator, SimpleDecorator
+                SimpleDecorator()
+                def test; 42 end
+            }
+        }
+
+        context 'Check decorated_method value' do
+            let(:call_proc) { proc { decorated_method } }
+            it { expect(instance.test).to eq TestClass.instance_method(:test) }
+        end
+
+        context 'Check undecorated_method value' do
+            let(:call_proc) { proc { undecorated_method } }
+            it { expect(instance.test.bind(instance).call).to eq 42 }
+            it { expect(instance.test.bind(instance).call).not_to eq TestClass.instance_method(:test) }
```

Comment:
`instance.test` returns the original method
`TestClass`.instance_method(:test)` returns the decorated method

They should not be the same